src/snowcli/output/decorators.py

```diff
@@ -1,14 +1,33 @@
 from __future__ import annotations
 
 from functools import wraps
-
-from snowcli.output.printing import print_db_cursor
+from snowcli.output.printing import OutputData, print_output
+from snowflake.connector.cursor import SnowflakeCursor
 
 
 def with_output(func):
     @wraps(func)
     def wrapper(*args, **kwargs):
         result = func(*args, **kwargs)
-        print_db_cursor(result)
+        output_data = OutputData()
+
+        if result is None:
+            print_output()
+            return wrapper
+
+        if _is_list_of_results(result):
+            for single_result in result:
+                output_data.add_data(single_result)
+        else:
+            output_data.add_data(result)
+        print_output(output_data)
 
     return wrapper
+
+
+def _is_list_of_results(result):
+    return (
+        isinstance(result, list)
+        and len(result) > 0
+        and (isinstance(result[0], list) or isinstance(result[0], SnowflakeCursor))
+    )
```

Comment:
I think we should support more general cases. The result should be iterable. But there's a lot of possibilites. So the question is - do we want to support generic types? Maybe we should always expect `OutputData`?

For example we can do
```py
return OutputData.from_cursor(result)

return OutputData.from_string("ok")

return OutputData.from_list([x for x in xs])

return OutputData.from_iterable(x for x in xs)
```

Each such implementation will have it's own checks and the only validation we do in `with_output` is to check if result is instance of `OutputData`.

Other thing to consider (in future work) is to extend OutputData  class with support for prompts. This would unify console output in 100%